pkg/action/action_handler.go

```diff
@@ -144,6 +168,32 @@ func (h *ActionHandler) ExecuteAction(actionExecutionDTO *proto.ActionExecutionD
 	return result, nil
 }
 
+func keepAlive(tracker sdkprobe.ActionProgressTracker, stop chan struct{}) {
```

Comment:
yes, we are using the UpdateProgress in this function keepAlive().  Currently, actionHandler does not pass the ActionProgressTracker down the ActionExecutor(rescheduler, Scaler). So the UpdateProgress() is called in this keepAlive() function.